src/common/create-ignorer.js

```diff
@@ -21,9 +31,20 @@ async function createIgnorer(ignorePath, withNodeModules) {
  * @param {boolean?} withNodeModules
  */
 createIgnorer.sync = function (ignorePath, withNodeModules) {
-  const ignoreContent = !ignorePath
+  let ignoreContent = !ignorePath
     ? null
     : getFileContentOrNull.sync(path.resolve(ignorePath));
+
+  if (!ignoreContent) {
+    try {
+      ignoreContent = JSON.parse(
+        getFileContentOrNull.sync(path.resolve("package.json"))
```

Comment:
I think we should search `package.json` from the file not `cwd`.

Reply:
Is something like [this](https://github.com/prettier/prettier/pull/12672/commits/7256153fb0d9bdfd609de8e87ec3111334c290e1) fine?

Reply:
No, not the ignore file, the formatting file.

Reply:
Maybe it should be done in `getFileInfo` function.

Reply:
Okay, how about [this](https://github.com/prettier/prettier/pull/12672/commits/3b0bd49b4b54ef23db2a85a58c8a9ba3db0508c9) then? I have to admit it's pretty ugly though.